### ISSUE TITLE
Yarn 0.17.0 unexpected manifest error in linux

### ISSUE THREAD

**Issue by iislucas** [yarnpkg/yarn#1855]
**bug**

**What is the current behavior?**

Install command fails with `error An unexpected error occurred: "expected manifest".` 

```
root@47f9e3db7100:/app# npm install -g --unsafe-perm yarn@0.17.0
root@47f9e3db7100:/app# yarn global add --ignore-optional \
>   pm2 \
>   protractor \
>   ts-node \
>   typescript@2.0.9 \
>   typings@1.5.0 \
>   webpack
yarn global v0.17.0
[1/4] Resolving packages...
warning protractor > jasmine > glob > minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
[2/4] Fetching packages...
warning fsevents@1.0.15: The platform "linux" is incompatible with this module.
info "fsevents@1.0.15" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
error An unexpected error occurred: "expected manifest".
info If you think this is a bug, please open a bug report with the information provided in "/root/.config/yarn/global/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/global for documentation about this command.
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Try to build this Dockerfile:
```
FROM gcr.io/google_appengine/nodejs

# Undo the env variable setting from the google nodejs env.
# We set NODE_ENV ourself when we run the server, rather than have a global
# setting which messes with npm install.
ENV NODE_ENV ''

# RUN curl https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
# RUN echo "deb http://dl.yarnpkg.com/debian/ stable main" | \
#      tee /etc/apt/sources.list.d/yarn.list

RUN apt-get -q update && \
    apt-get install --no-install-recommends -y -q \
      nano less memcached rsync vim
#      yarn=0.16.1

# Make sure memcache has started.
RUN memcached -d -u root

# Node, npm, yarn setup
RUN install_node 7.1

# Note: because this is running as super-user in docker, we need to specify the
# --unsafe-perm flag to allow npm not to worry about downgrading its
# permissions.
#
# TODO: 0.17.0 is broken for linux.
# RUN npm remove -g --unsafe-perm yarn@0.17.0
RUN npm install -g --unsafe-perm yarn@0.16.1

# `pm2` runs the production server, retsarting if it crashes, etc.
# `protractor` is used for testing angular2 components.
# `ts-node` allows typesscipt scripts to executed inline, like node/js.
# `typescript` is the programming language and compiler for javascript.
# `typings` provides type definitions for typescript needed to compile code.
# `webpack` is used to combine/bundle the HTML/CSS/JS, produce source maps, etc.
RUN yarn global add --ignore-optional \
  pm2 \
  protractor \
  ts-node \
  typescript@2.0.9 \
  typings@1.5.0 \
  webpack
```

Which can be done by putting that into a file called `Dockerfile`, and running `docker build .`. Or by using a fresh installed ubuntu and simply running the above commands (obviously without the `RUN`)

**What is the expected behavior?**

The packages are successfully installed.

**Please mention your node.js, yarn and operating system version.**

Docker/Linux (running using virtual box on MacOS 10.11.6): 

```
$ uname -a 
Linux 0361e16172da 4.1.19-boot2docker #1 SMP Mon Mar 7 17:44:33 UTC 2016 x86_64 GNU/Linux
```

Linux Distribution (within docker)

```
root@0361e16172da:/app# cat /etc/*-release
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

Note version: 

```
root@0361e16172da:/app# node -v
v7.1.0
```

**Comment by Daniel15** [comment]
The Debian repo was rolled back to 0.16.1... Try installing through that (you've got it commented out) rather than npm.

**Comment by markstos** [comment]
Why the rollback?

**Comment by wyze** [comment]
Duplicate of #1824 and fixed via #1840.

**Comment by Daniel15** [comment]
There's a bunch of issues with 0.17.0 so we temporarily rolled back to 0.16.1 until fixes are submitted.
